src/main/scala/cromwell/engine/backend/jes/JesAttributes.scala

```diff
@@ -0,0 +1,41 @@
+package cromwell.engine.backend.jes
+
+import java.net.URL
+
+import com.typesafe.config.{Config, ConfigFactory}
+import cromwell.util.ConfigUtil._
+
+case class JesAttributes(applicationName: String,
+                         project: String,
+                         executionBucket: String,
+                         endpointUrl: URL,
+                         authMode: GcsAuthMode,
+                         docker: Option[DockerAuthInformation])
+object JesAttributes {
+
+  def apply(): JesAttributes = {
+    val jesConf = ConfigFactory.load.getConfig("backend").getConfig("jes")
+    val refConf: Config = ConfigFactory.parseResources("jes.conf")
+
+    jesConf.checkValidWrapped(refConf, "Jes")
+    jesConf.warnNotRecognized(refConf, "Jes", "dockerAccount", "dockerToken")
+
+    val applicationName = jesConf.getString("applicationName")
+    val project = jesConf.getString("project")
+    val executionBucket = jesConf.getString("baseExecutionBucket")
+    val endpointUrl = jesConf.getURL("endpointUrl")
+    val authMode = GcsAuthMode.fromString(jesConf.getString("authenticationMode"))
+    val docker = for {
+      account <- jesConf.getStringOption("dockerAccount")
+      token <- jesConf.getStringOption("dockerToken")
+    } yield DockerAuthInformation(account, token)
+
```

Comment:
I totally agree w/ what you're saying in that it conveniently removes the need for code on our side. What's making my spidey sense tingle is it seems that it then adds the need to slap on special case code here and there which ends up being a bit confusing. 

Now I'm really just thinking out loud but perhaps the right thing is to break the JES stuff up a bit further so instead of say a jes backend w/ maybe a docker auth it's either a jes backend or a jes backend w/ a docker auth. I def haven't thought that through any longer than it took to type it out thoug h:)